pkg/webui/console/store/middleware/logics/devices.js

```diff
@@ -128,11 +94,25 @@ const getDeviceTemplateFormatsLogic = createRequestLogic({
   },
 })
 
+const getDeviceSessionLogic = createLogic({
+  type: devices.GET_DEVICE_EVENT_MESSAGE_SUCCESS,
+  process: async ({ action }, dispatch, done) => {
+    const { event, id } = action
+    const appId = id.application_ids.application_id
+
+    if (event.name === 'as.up.join.forward') {
+      dispatch(devices.getDevice(appId, id.device_id, ['session']))
+      done()
+    }
```

Comment:
I think it would be nice to actually send a toast notification when the device joins as well. You can dispatch the toast from within the logic:
```js
        toast({
          title: id.device_id,
          message: m.joinSuccess,
          type: toast.types.INFO,
        })
```
> The device has successfully joined the network